Automattic/jetpack
# Search: Migrate more helper classes to package

Split off from #21637.

#### Changes proposed in this Pull Request:
* Migrates the template tags helper class, which is primarily responsible for rendering clickable filters within the Jetpack Search widget.
* Migrates the site settings helper class, which is used to persist changes from Customberg onto site options. 

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Apply this change to your site with a Jetpack Search subscription.
2. Add the Jetpack Search widget with configured filters to a site sidebar with only the search module enabled (i.e. instant search disabled).
3. Ensure that the filters from (2) render as expected.
4. Repeat (2) and (3) with Instant Search enabled in addition to the search module.
5. Navigate to Customberg (`/wp-admin/admin.php?page=jetpack-search-configure`).
6. Ensure Customberg reads and saves instant search configuration settings as expected.
